### ISSUE TITLE
Generate Terraform Import Code for TFM Created Resources

### ISSUE THREAD

**Issue by jeffmccollum** [hashicorp-services/tfm#135]
In the destination environment after TFM has completed a migration, new workspaces, teams, and variable sets will have been created. While this is great for the migration, this leaves Day 2 operations in a more difficult spot, as these resources are not being managed by Terraform. 

It would be great if TFM could generate Terraform import statements for all resources that TFM Creates during a migration!

**Comment by drewmullen** [comment]
Is the idea here to generate the `tfe_` resources? that is a really good idea 

maybe instead of generating import statements you could generate [import blocks](https://developer.hashicorp.com/terraform/language/import#syntax)

**Comment by jeffmccollum** [comment]
Yep, goal would be to auto generate these resources, `tfe_workspace`, `tfe_variable_set`, `tfe_team` and others that tfm would create. 
Now the question that we are still thinking, is for generation of the terraform code, we could build the raw resources, 1 to 1 or we could do something like building a variable map and call a module (https://registry.terraform.io/modules/alexbasista/workspacer/tfe/latest) with a for each. What would you like to see?
Yes, definitely using the latest import blocks, use to old import method :)

**Comment by drewmullen** [comment]
@jeffmccollum we're going to start using and recommending `workspacer` so i agree - push forward with that one!

**Comment by abuxton** [comment]
look at tf-migrate this idea was implemented in that source code

**Comment by abuxton** [comment]
## Implementation Analysis & Expanded Proposal

Picking up from the earlier discussion — the `workspacer` direction is the right call. Here is a fuller breakdown of what this would look like as a concrete implementation inside `tfm`.

---

### What `tfm copy` currently creates on the destination

Based on the current codebase, the following TFE resources are created by `tfm` during a migration and **none of them are managed by Terraform after the fact**:

| `tfm` command | TFE resource created | Terraform provider resource |
|---|---|---|
| `tfm copy workspaces` | Workspaces, Projects | `tfe_workspace`, `tfe_project` |
| `tfm copy teams` | Teams | `tfe_team` |
| `tfm copy varsets` | Variable Sets + Variables | `tfe_variable_set`, `tfe_variable_set_variable` |
| `tfm copy workspaces-vars` | Workspace Variables | `tfe_variable` |
| `tfm copy workspaces-teamaccess` | Team Access bindings | `tfe_team_access` |
| `tfm copy workspaces-run-triggers` | Run Triggers | `tfe_workspace_run_task` |
| `tfm core create-workspaces` | Workspaces (CE migration) | `tfe_workspace` |

---

### Proposed: `tfm generate import`

Add a new subcommand under the existing `generate` command:

```
tfm generate import [flags]
```

**Flags:**

| Flag | Description |
|---|---|
| `--resources` | Comma-separated list: `workspaces,teams,varsets,projects` (default: all) |
| `--output` | Output file path (default: `tfm-imports.tf`) |
| `--format` | `blocks` (default, Terraform 1.5+ import blocks) or `cli` (legacy `terraform import` commands) |
| `--module` | Wrap resources in the `workspacer` module pattern instead of flat resources |

---

### Output: Terraform 1.5+ import blocks (default)

```hcl
# Generated by tfm generate import
# Destination org: my-org  |  Generated: 2024-01-15T10:30:00Z

import {
  to = tfe_workspace.ws_my_workspace
  id = "my-org/my-workspace"
}

import {
  to = tfe_project.proj_platform
  id = "prj-AbCd1234"
}

import {
  to = tfe_team.team_developers
  id = "team-XyZw5678"
}

import {
  to = tfe_variable_set.varset_aws_creds
  id = "varset-Mn0p9012"
}
```

---

### Output: `--module workspacer` variant

When `--module` is set, generate a `workspacer` module call using a `for_each` local map — suitable for teams that want to manage their org via the [`alexbasista/workspacer/tfe`](https://registry.terraform.io/modules/alexbasista/workspacer/tfe/latest) module:

```hcl
# Generated by tfm generate import --module workspacer

locals {
  workspaces = {
    "my-workspace" = {
      organization   = "my-org"
      vcs_repo_identifier = "my-org/my-repo"
      terraform_version   = "1.6.0"
      tag_names           = ["team:platform"]
    }
    "another-workspace" = {
      organization = "my-org"
      terraform_version = "1.5.0"
    }
  }
}

module "workspaces" {
  source   = "alexbasista/workspacer/tfe"
  version  = "~> 0.9"

  for_each     = local.workspaces
  organization = each.value.organization
  workspace_name = each.key
  # ... remaining attributes
}
```

---

### Implementation notes

1. **Where to add it**: `./cmd/generate/` already exists with `generate config`. A new file `./cmd/generate/import.go` keeps the pattern clean.

2. **Source of truth**: The command hits the **destination** TFE/TFC API (using `tfclient.GetDestinationClientContexts()`) and lists resources to build the import manifest. No local state or cache file needed.

3. **Naming convention for resource addresses**: Sanitise workspace/team/project names — replace spaces and special chars with `_`, prefix with the resource type short name (e.g. `ws_`, `team_`, `proj_`, `varset_`).

4. **IDs**: 
   - `tfe_workspace` id = `"<org>/<workspace-name>"`
   - `tfe_project` id = internal project ID (e.g. `prj-xxxx`)
   - `tfe_team` id = internal team ID (e.g. `team-xxxx`)
   - `tfe_variable_set` id = internal varset ID (e.g. `varset-xxxx`)

5. **Companion resource stubs** (optional `--with-resources` flag): emit skeleton `resource` blocks alongside the `import` blocks so `terraform plan` passes immediately after `terraform init`.

6. **Filtering**: respect the existing `workspaces` list in `.tfm.hcl` so only the migrated subset is exported — not the entire org.

---

### Minimal viable scope for a first PR

A focused first pass could:
- Add `tfm generate import` with `--format blocks` (Terraform 1.5+ import blocks)
- Cover `tfe_workspace`, `tfe_project`, `tfe_team`, `tfe_variable_set`
- Write output to `tfm-imports.tf` by default
- Use `--resources` flag to scope output

The `--module workspacer` path and `--with-resources` stubs can follow in subsequent PRs.

---

> **Note on prior art**: the `tf-migrate` source referenced in the last comment has patterns for generating HCL from API data that can be adapted here. The existing `generate config` command in `./cmd/generate/config.go` shows the `html/template` + file-write pattern already in use in `tfm`.
